Lib/Protocols/IdFTP.pas

```diff
@@ -3988,7 +3989,10 @@ function TIdFTPClientIdentifier.GetClntOutput: String;
 
 function TIdFTPClientIdentifier.GetCSIDOutput: String;
 begin
-  Result := 'Name='+FClientName;
+  Result := '';
+  if FClientName <> '' then begin
+    Result := 'Name='+FClientName;
+  end;
   if FClientVersion <> '' then begin
     Result := Result + '; Version='+FClientVersion;

```

Comment:
If `FClientName` is empty, but `FClentVersion` and/or `FVendor` are not empty, then `GetCSIDOutput()` will return invalid output.